### PR TITLE
Adding sound and monster ai

### PR DESCRIPTION
**Pull request by JackAkers**
Added sound system and background music.  Also has monster AI and lose condition.

### CODE REVIEW

File: include/GameView.h [hunk 1]
```diff
@@ -3,8 +3,14 @@
 #include <SFML/Graphics.hpp>
 #include <SFML/Audio.hpp>
 #include "InputManager.h"
+//<<<<<<< HEAD
+#include "MonsterAI.h"
+//=======
```
Comment (alexchung1233) [inline include/GameView.h:8]: remove these merge conflict indicators

File: include/GameView.h [hunk 2]
```diff
@@ -3,8 +3,14 @@
...
 #include "State.h"
 #include "GameOver.h"
+#include "Sound.h"
+
+//>>>>>>> main
```
Comment (alexchung1233) [inline include/GameView.h:13]: remove this

File: include/Menu.h [hunk 3]
```diff
@@ -11,7 +12,7 @@ class Menu : public State
   public:
     //Menu();
 
-    Menu(sf::RenderWindow& app);
+    Menu(sf::RenderWindow& app, Sound *sound);
```
Comment (alexchung1233) [inline include/Menu.h:15]: we shouldn't need to pass a sound into the constructor

File: src/GameView.cpp [hunk 4]
```diff
@@ -62,15 +81,37 @@ void GameView::update(sf::Event& Event, float dt){
...
+
+  float distX = pow(monsterAI.positionX - player.getPosition().x-125, 2);
+  float distY = pow(monsterAI.positionY - player.getPosition().y+20, 2);
+
+
+  if (sqrt(distX + distY) < 70){
+    this->status = State::SUCCESS;
+    childState = new GameOver(*App, "You Lose...", sound);
+  }
+//=======
```
Comment (alexchung1233) [inline src/GameView.cpp:102]: remove this

File: src/GameView.cpp [hunk 5]
```diff
@@ -62,15 +81,37 @@ void GameView::update(sf::Event& Event, float dt){
   PlayerActor player = this->logic.getPlayer();
   sprite_player.setPosition(player.getPosition().x, player.getPosition().y);
 
+//<<<<<<< HEAD
```
Comment (alexchung1233) [inline src/GameView.cpp:84]: remove this

File: src/Menu.cpp [hunk 6]
```diff
@@ -4,11 +4,12 @@
 #include <string>
 #include <iostream>
 
-Menu::Menu(sf::RenderWindow& app){
+Menu::Menu(sf::RenderWindow& app, Sound* newSound){
```
Comment (alexchung1233) [inline src/Menu.cpp:7]: shouldn't pass a sound the constructor

File: src/Menu.cpp [hunk 7]
```diff
@@ -173,8 +176,10 @@ void Menu::update(sf::Event& Event, float dt){
 
       int num = enterPressed();
       if(num == 0){
+        sound->stopMenuMusic();
+        sound->playPlayingMusic();
         this->status = State::SUCCESS;
-        this->childState = new GameView(*App);
+        this->childState = new GameView(*App, sound);
```
Comment (alexchung1233) [inline src/Menu.cpp:182]: shouldn't pass through GameView

File: include/Menu.h (hunk 3 above)
Comment (alexchung1233) [reply include/Menu.h:15]: I recommend making a setSound method inside State.h and having GameView, Menu, all the other states inherit from it.

File: src/GameManager.cpp [hunk 8]
```diff
@@ -5,6 +5,7 @@
 
 
 GameManager::GameManager(){
+  sound.init();
```
Comment (alexchung1233) [inline src/GameManager.cpp:8]: this should be in one of the other functions

**Review by alexchung1233 — CHANGES_REQUESTED** [review]

File: src/Sound.cpp [hunk 9]
```diff
@@ -0,0 +1,71 @@
+#include <iostream>
+#include "Sound.h"
+
+Sound::Sound(){
+
```
Comment (alexchung1233) [inline src/Sound.cpp:5]: you should make a generic Sound constructor that takes in a file and lets you play it. Right now if you want to add a lightning sound, for instance, you have to make another function

**Review by alexchung1233 — CHANGES_REQUESTED** [review]

**Review by ktkel — APPROVED** [review]

**Review by alexchung1233 — APPROVED** [review]

**Pull request merged**